superset-frontend/spec/javascripts/dashboard/util/isValidChild_spec.ts

```diff
@@ -136,20 +136,22 @@ describe('isValidChild', () => {
     invalidExamples.forEach((example, exampleIdx) => {
       let childDepth = 0;
       example.forEach((childType, i) => {
-        const shouldTestChild = Array.isArray(childType);
-
-        if (i > 0 && shouldTestChild) {
+        // should test child
+        if (i > 0 && Array.isArray(childType)) {
           const parentDepth = childDepth - 1;
           const parentType = example[i - 1];
 
+          if (typeof parentType !== 'string')
+            throw TypeError('parent must be string');
+
           it(`(${exampleIdx})${getIndentation(
             childDepth,
           )}${parentType} (depth ${parentDepth}) > ${childType} ❌`, () => {
             expect(
               isValidChild({
                 parentDepth,
                 parentType,
-                childType,
+                childType: childType[0],
```

Comment:
this wasn't quite testing what it said it was testing, it was seeing if an array matches the string literal at a given level.
This fixes the tests to compare strings to strings.